Xnnpack test for program-data separation

lucylq opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #10532

Add xnnpack test for program-data separation

Differential Revision: [D73794695](https://our.internmc.facebook.com/intern/diff/D73794695/)

lucylq committed:
Xnnpack test for program-data separation

Add xnnpack test for program-data separation

Differential Revision: [D73794695](https://our.internmc.facebook.com/intern/diff/D73794695/)

[ghstack-poisoned]

digantdesai commented on backends/xnnpack/test/targets.bzl:
```diff
@@ -43,3 +43,23 @@ def define_common_targets():
+                # Uses an fbcode target path because the authoring/export tools
+                # intentionally don't work in xplat (since they're host-only
+                # tools).
+                "ET_MODULE_LINEAR_XNN_PROGRAM_PATH": "$(location fbcode//executorch/test/models:exported_xnnpack_program_and_data[ModuleLinear-e.pte])",
```
can this be an e2e test? instead of this?

lucylq replied:
Hmn, could you expand a bit more? It should test the E2E

lucylq committed:
Update on "Xnnpack test for program-data separation"


Add xnnpack test for program-data separation

Differential Revision: [D73794695](https://our.internmc.facebook.com/intern/diff/D73794695/)

[ghstack-poisoned]

lucylq committed:
Update on "Xnnpack test for program-data separation"


Add xnnpack test for program-data separation

Differential Revision: [D73794695](https://our.internmc.facebook.com/intern/diff/D73794695/)

[ghstack-poisoned]

lucylq committed:
Update on "Xnnpack test for program-data separation"


Add xnnpack test for program-data separation

Differential Revision: [D73794695](https://our.internmc.facebook.com/intern/diff/D73794695/)

[ghstack-poisoned]

lucylq committed:
Update on "Xnnpack test for program-data separation"


Add xnnpack test for program-data separation

Differential Revision: [D73794695](https://our.internmc.facebook.com/intern/diff/D73794695/)

[ghstack-poisoned]
